src/BenchmarksApps/HttpClientBenchmarks/Clients/HttpClient/Program.cs

```diff
@@ -0,0 +1,321 @@
+        var stopwatch = Stopwatch.StartNew();
+        var response = await s_httpClients[0].SendAsync(request, CancellationToken.None);
+        var elapsed = stopwatch.ElapsedMilliseconds;
+        response.EnsureSuccessStatusCode();
+        BenchmarksEventSource.Register("http/firstrequest", Operations.Max, Operations.Max, "First request duration (ms)", "Duration of the first request to the server (ms)", "n0");
+        LogMetric("http/firstrequest", elapsed);
+    }
+
+    private static async Task RunScenario()
+    {
+        BenchmarksEventSource.Register("http/successrequests", Operations.Sum, Operations.Sum, "Success Requests", "Number of successful requests", "n0");
+        BenchmarksEventSource.Register("http/badstatusrequests", Operations.Sum, Operations.Sum, "Bad Status Code Requests", "Number of requests with bad status codes", "n0");
+        BenchmarksEventSource.Register("http/exceptions", Operations.Sum, Operations.Sum, "Exceptions", "Number of exceptions", "n0");
+        BenchmarksEventSource.Register("http/rps/mean", Operations.Avg, Operations.Avg, "Mean RPS", "Requests per second - mean", "n0");
+
+        if (s_options.CollectRequestTimings)
+        {
+            RegisterPercentiledMetric("http/headers", "Time to headers (ms)", "Time to headers (ms)");
+            RegisterPercentiledMetric("http/contentstart", "Time to first content byte (ms)", "Time to first content byte (ms)");
+            RegisterPercentiledMetric("http/contentend", "Time to last content byte (ms)", "Time to last content byte (ms)");
+        }
+
+        Func<HttpMessageInvoker, Task<Metrics>> scenario;
+        switch(s_options.Scenario)
+        {
+            case "get":
+                scenario = Get;
+                break;
+            default:
+                throw new ArgumentException($"Unknown scenario: {s_options.Scenario}");
+        }
+
+        s_isWarmup = true;
+        s_isRunning = true;
+
+        var coordinatorTask = Task.Run(async () =>
+        {
+            await Task.Delay(TimeSpan.FromSeconds(s_options.Warmup));
+            s_isWarmup = false;
+            Log("Completing warmup...");
+            await Task.Delay(TimeSpan.FromSeconds(s_options.Duration));
+            s_isRunning = false;
+            Log("Completing scenario...");
+        });
+
+        var tasks = new List<Task<Metrics>>(s_options.NumberOfHttpClients * s_options.ConcurrencyPerHttpClient);
+        for (int i = 0; i < s_options.NumberOfHttpClients; ++i)
+        {
+            var client = s_httpClients[i];
+            for (int j = 0; j < s_options.ConcurrencyPerHttpClient; ++j)
+            {
+                tasks.Add(scenario(client));
+            }
+        }
+
+        await coordinatorTask;
+        var metricsArray = await Task.WhenAll(tasks);
+        foreach (var metrics in metricsArray)
+        {
+            s_metrics.Add(metrics);
+        }
+
+        LogMetric("http/successrequests", s_metrics.SuccessRequests);
```

Comment:
Would be helpful if it could use the same metrics as the other clients, because I can then mix the values from different clients in the charts on PowerBI.

List from wrk for instance:

```
http/rps/mean
http/requests
http/latency/mean
http/latency/max
http/requests/badresponses
http/requests/errors
http/throughput
http/latency/50, http/latency/75, http/latency/90, http/latency/99
```